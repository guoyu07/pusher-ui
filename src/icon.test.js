import React from 'react';
import renderer from 'react-test-renderer';

import Icon from './icon';

describe('Icon', () => {
  const createSnapshot = props => {
    expect(
      renderer.create(<Icon name="account" {...props} />).toJSON(),
    ).toMatchSnapshot();
  };

  it('should render an icon from the icon set with default props', () => {
    createSnapshot({ name: 'account' });
    createSnapshot({ name: 'pusher-logo' });
  });

  it('should not render anything if no icon is found', () => {
    console.warn = jest.fn(); // eslint-disable-line no-console
    createSnapshot({ name: 'not-really-an-icon' });
  });

  it('should allow configuring its size', () => {
    createSnapshot({ size: 12 });
    createSnapshot({ size: '24' });
    createSnapshot({ size: '16x18' });
  });

  it('should allow configuring its color', () => {
    createSnapshot({ color: 'red' });
  });

  it('should pass on any css props', () => {
    createSnapshot({ padding: 10 });
  });
});
