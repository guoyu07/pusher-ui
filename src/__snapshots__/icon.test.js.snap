// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should allow configuring its color 1`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 24px;
  min-height: 24px;
}

<svg
  className="glamor-0"
  color="red"
  height={24}
  width={24}
>
  <use
    xlinkHref="#icon:account"
  />
</svg>
`;

exports[`Icon should allow configuring its size 1`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 12px;
  min-height: 12px;
}

<svg
  className="glamor-0"
  color="inherit"
  height={12}
  width={12}
>
  <use
    xlinkHref="#icon:account"
  />
</svg>
`;

exports[`Icon should allow configuring its size 2`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 24px;
  min-height: 24px;
}

<svg
  className="glamor-0"
  color="inherit"
  height="24"
  width="24"
>
  <use
    xlinkHref="#icon:account"
  />
</svg>
`;

exports[`Icon should allow configuring its size 3`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 16px;
  min-height: 18px;
}

<svg
  className="glamor-0"
  color="inherit"
  height="18"
  width="16"
>
  <use
    xlinkHref="#icon:account"
  />
</svg>
`;

exports[`Icon should not render anything if no icon is found 1`] = `null`;

exports[`Icon should pass on any css props 1`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 24px;
  min-height: 24px;
  padding: 10px;
}

<svg
  className="glamor-0"
  color="inherit"
  height={24}
  width={24}
>
  <use
    xlinkHref="#icon:account"
  />
</svg>
`;

exports[`Icon should render an icon from the icon set with default props 1`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 24px;
  min-height: 24px;
}

<svg
  className="glamor-0"
  color="inherit"
  height={24}
  width={24}
>
  <use
    xlinkHref="#icon:account"
  />
</svg>
`;

exports[`Icon should render an icon from the icon set with default props 2`] = `
.glamor-0,
[data-glamor-0] {
  fill: currentcolor;
  vertical-align: middle;
  min-width: 24px;
  min-height: 24px;
}

<svg
  className="glamor-0"
  color="inherit"
  height={24}
  width={24}
>
  <use
    xlinkHref="#icon:pusher-logo"
  />
</svg>
`;
